jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFiles.java

```diff
@@ -111,40 +102,20 @@ public static JibContainerBuilder toJibContainerBuilder(
     return containerBuilder;
   }
 
-  // TODO: add testing, need to do via intergration tests as there's no good way to extract out that
-  //   the base image was populated as the user intended currently.
-  static JibContainerBuilder createJibContainerBuilder(
-      BaseImageSpec from, CommonCliOptions commonCliOptions, ConsoleLogger logger)
+  private static JibContainerBuilder createJibContainerBuilder(
+      BaseImageSpec baseImageSpec, CommonCliOptions commonCliOptions, ConsoleLogger logger)
       throws InvalidImageReferenceException, FileNotFoundException {
-    String baseImageReference = from.getImage();
-    if (baseImageReference.startsWith(DOCKER_DAEMON_IMAGE_PREFIX)) {
-      return Jib.from(
-          DockerDaemonImage.named(baseImageReference.replaceFirst(DOCKER_DAEMON_IMAGE_PREFIX, "")));
-    }
-    if (baseImageReference.startsWith(TAR_IMAGE_PREFIX)) {
-      return Jib.from(
-          TarImage.at(Paths.get(baseImageReference.replaceFirst(TAR_IMAGE_PREFIX, ""))));
-    }
-    ImageReference imageReference =
-        ImageReference.parse(baseImageReference.replaceFirst(REGISTRY_IMAGE_PREFIX, ""));
-    RegistryImage registryImage = RegistryImage.named(imageReference);
-    DefaultCredentialRetrievers defaultCredentialRetrievers =
-        DefaultCredentialRetrievers.init(
-            CredentialRetrieverFactory.forImage(
-                imageReference,
-                logEvent -> logger.log(logEvent.getLevel(), logEvent.getMessage())));
-    Credentials.getFromCredentialRetrievers(commonCliOptions, defaultCredentialRetrievers)
-        .forEach(registryImage::addCredentialRetriever);
-    JibContainerBuilder containerBuilder = Jib.from(registryImage);
-    if (!from.getPlatforms().isEmpty()) {
-      containerBuilder.setPlatforms(
-          from.getPlatforms()
+    List<Platform> platforms = new ArrayList<>();
+    if (!baseImageSpec.getPlatforms().isEmpty()) {
```

Comment:
This `if (...)` probably isn't necessary. For an empty list, stream will return an empty collection. So just `List<Platform> platforms = baseImageSpec.getPlatforms().stream().map().collect()`.